### PR TITLE
Enable atlas generation on reporting

### PR DESCRIPTION
**Pull request by myarjunar**
### What does it fix?
* Ticket: #4303 #3059 
* Funded by: DFAT
* Description: Enable atlas generation.

selection_121
[a4-landscape-blue.pdf](https://github.com/inasafe/inasafe/files/1197436/a4-landscape-blue.pdf)

selection_120
[a4-landscape-blue.pdf](https://github.com/inasafe/inasafe/files/1197439/a4-landscape-blue.pdf)

### CODE REVIEW

File: safe/report/processors/default.py [hunk 1]
```diff
@@ -592,3 +605,114 @@ def qt_svg_to_png_renderer(impact_report, component):
...
+            composition.beginPrint(printer)
+            if not painter.begin(printer):
+                msg = 'Atlas processing error: ' \
+                      'Cannot write to {output}.'.format(output=output_path)
+                LOGGER.error(msg)
+                return
+
+        LOGGER.info('Exporting Atlas')
+
+        atlas_output= []
```
Comment (ismailsunni) [inline safe/report/processors/default.py]: __[flake8]__
* [E225] missing whitespace around operator

File: safe/report/processors/default.py [hunk 2]
```diff
@@ -592,3 +605,114 @@ def qt_svg_to_png_renderer(impact_report, component):
...
+                multi_file_printer = QPrinter(QPrinter.HighResolution)
+                current_filename = atlas_composition.currentFilename()
+                output_path = os.path.join(
+                    output_directory, current_filename + '.pdf')
+                composition.beginPrintAsPDF(multi_file_printer, output_path)
+                composition.beginPrint(multi_file_printer)
+                if not painter.begin(multi_file_printer):
+                    msg = 'Atlas processing error: ' \
+                          'Cannot write to {output}.'.format(
+                        output=output_path)
```
Comment (ismailsunni) [inline safe/report/processors/default.py]: __[flake8]__
* [E122] continuation line missing indentation or outdented

File: safe/report/processors/default.py [hunk 3]
```diff
@@ -592,3 +605,114 @@ def qt_svg_to_png_renderer(impact_report, component):
 
     component.output = output_path
     return component.output
+
+
+def atlas_renderer(composition, coverage_layer, output_path, file_format):
+    """Extract composition using atlas generation.
+
+    :param composition: QGIS Composition object
```
Comment (ismailsunni) [inline safe/report/processors/default.py]: dot in the end of sentence :), same for the others

File: safe/report/processors/default.py [hunk 4]
```diff
@@ -592,3 +605,114 @@ def qt_svg_to_png_renderer(impact_report, component):
...
+            if not atlas_on_single_file:
+                # we need another printer object fot multi file atlas
+                multi_file_printer = QPrinter(QPrinter.HighResolution)
+                current_filename = atlas_composition.currentFilename()
+                output_path = os.path.join(
+                    output_directory, current_filename + '.pdf')
+                composition.beginPrintAsPDF(multi_file_printer, output_path)
+                composition.beginPrint(multi_file_printer)
+                if not painter.begin(multi_file_printer):
+                    msg = 'Atlas processing error: ' \
```
Comment (ismailsunni) [inline safe/report/processors/default.py]: Avoid backslash if possible

File: safe/report/processors/default.py [hunk 5]
```diff
@@ -592,3 +605,114 @@ def qt_svg_to_png_renderer(impact_report, component):
...
+                      'Cannot write to {output}.'.format(output=output_path)
+                LOGGER.error(msg)
+                return
+
+        LOGGER.info('Exporting Atlas')
+
+        atlas_output = []
+        for feature_index in range(atlas_composition.numFeatures()):
+            if not atlas_composition.prepareForFeature(feature_index):
+                msg = 'Atlas processing error: Exporting atlas error at ' \
```
Comment (ismailsunni) [inline safe/report/processors/default.py]: Use brackets to avoid backslash

File: safe/report/processors/default.py [hunk 6]
```diff
@@ -592,3 +605,114 @@ def qt_svg_to_png_renderer(impact_report, component):
...
+                error=atlas_composition.featureFilterErrorString())
+            LOGGER.error(msg)
+            return
+
+        if atlas_on_single_file:
+            atlas_composition.prepareForFeature(0)
+            composition.beginPrintAsPDF(printer, output_path)
+            composition.beginPrint(printer)
+            if not painter.begin(printer):
+                msg = 'Atlas processing error: ' \
```
Comment (ismailsunni) [inline safe/report/processors/default.py]: Avoid backslash

File: safe/gui/ui/options_dialog_base.ui [hunk 7]
```diff
@@ -252,18 +241,36 @@
...
+                <widget class="QCheckBox" name="cbx_show_only_impact">
+                 <property name="text">
+                  <string>Show only impact layer on report map</string>
+                 </property>
+                </widget>
+               </item>
+               <item row="7" column="0">
+                <widget class="QCheckBox" name="cbx_print_atlas_report">
+                 <property name="text">
+                  <string>Print atlas report on atlas driven template</string>
```
Comment (ismailsunni) [inline safe/gui/ui/options_dialog_base.ui]: Please make sure the string in the .ui file is set to translatable

Comment (myarjunar) [inline safe/gui/ui/options_dialog_base.ui]: yes it is

File: safe/report/processors/default.py [hunk 8]
```diff
@@ -592,3 +605,113 @@ def qt_svg_to_png_renderer(impact_report, component):
...
+    :param coverage_layer: Coverage Layer used for atlas map
+    :type coverage_layer: QgsMapLayer
+
+    :param output_path: the output path of the product
+    :type output_path: str
+
+    :param file_format: file format of map output, PDF or PNG
+    :type file_format: 'pdf', 'png'
+
+    :return: generated output path(s)
```
Comment (Gustry) [inline safe/report/processors/default.py]: We usually put capital letter and dot for `:return:` and `:param:` as @ismailsunni said ;-)

Comment (ismailsunni) [inline safe/report/processors/default.py]: Btw, I think I usually use `:returns:`. Not sure though which one is the right one :D

File: safe/report/processors/default.py [hunk 9]
```diff
@@ -592,3 +605,113 @@ def qt_svg_to_png_renderer(impact_report, component):
...
+    :type composition: qgis.core.QgsComposition
+
+    :param coverage_layer: Coverage Layer used for atlas map
+    :type coverage_layer: QgsMapLayer
+
+    :param output_path: the output path of the product
+    :type output_path: str
+
+    :param file_format: file format of map output, PDF or PNG
+    :type file_format: 'pdf', 'png'
```
Comment (ismailsunni) [inline safe/report/processors/default.py]: I just realised, is it a correct way to do enumeration?

Comment (myarjunar) [inline safe/report/processors/default.py]: It supposed to be str

File: safe/report/processors/default.py (hunk 8 above)
Comment (myarjunar) [inline safe/report/processors/default.py]: done. I don't know why I'm interpreting `dot in the end of sentence :), same for the others` as `give a complete sentence`

File: safe/gui/tools/help/options_help.py [hunk 10]
```diff
@@ -122,6 +122,13 @@ def content():
            'current project except for the impact layer. ')))
     bullets.add(m.Text(
         m.ImportantText(tr(
+            'Print atlas report on atlas driven template with the '
+            'aggregation layer')),
+        tr('When this option is enabled, InaSAFE will generate an atlas report'
```
Comment (Gustry) [inline safe/gui/tools/help/options_help.py]: You forgot one space at the end of your line. Be careful next time

**Pull request merged**